### PR TITLE
Added job FAILED_TO_RUN status. Changed the multi-run run_number format.

### PR DESCRIPTION
**Pull request by yhwen**

### CODE REVIEW

**Review by YuanTingHsieh — CHANGES_REQUESTED** [review]
Can you fix the unit test? Thanks

**Comment by yhwen** [comment]
> 

Fixed the extra space.

**Review by YuanTingHsieh — CHANGES_REQUESTED** [review]
suggest change one logic for clarity

File: nvflare/private/fed/server/job_runner.py [hunk 1]
```diff
@@ -51,14 +51,20 @@ def _deploy_job(self, job: Job, sites: list, fl_ctx: FLContext) -> str:
...
         count = 1
         while os.path.exists(workspace):
-            work_folder = run_number + "_" + str(count)
-            workspace = os.path.join(self.workspace_root, WorkspaceConstants.WORKSPACE_PREFIX + work_folder)
+            parts = run_number.split(":")
+            if len(parts) == 1:
+                run_number = run_number + ":" + str(count)
+            else:
+                run_number = parts[0] + ":" + str(count)
+            workspace = os.path.join(self.workspace_root, WorkspaceConstants.WORKSPACE_PREFIX + run_number)
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/private/fed/server/job_runner.py]: optimized the implementation.

File: nvflare/private/fed/server/job_runner.py [hunk 2]
```diff
@@ -161,6 +163,32 @@ def _stop_run(self, run_number, fl_ctx: FLContext):
...
+        message = Message(topic=TrainingTopic.DELETE_RUN, body="")
+        message.set_header(RequestHeader.RUN_NUM, str(run_number))
+        self.log_debug(fl_ctx, f"Send delete_run command to the site for run:{run_number}")
+        replies = self._send_to_clients(admin_server, client_sites, engine, message)
+        if not replies:
+            self.log_error(fl_ctx, f"Failed to send delete_run command to clients for run_{run_number}")
+
+        err = engine.delete_run_number(run_number)
+        if err:
+            self.log_error(fl_ctx, f"Failed to delete_run the server for run_.{run_number}")
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/job_runner.py:190]: Suggest these error messages should be change to

```f"Failed to XXX for run folder: {run_number}```

Comment (YuanTingHsieh) [reply nvflare/private/fed/server/job_runner.py:190]: Or if there are other places using this "run_" pattern we should also change them.

**Review by YuanTingHsieh — APPROVED** [review]
Minor improvements can be made on the error messages.

**Pull request merged**